### PR TITLE
Internal: Adjusted AI features to support updated WordPress UI v6.6 [ED-15387]

### PR DESCRIPTION
**Pull request by DennisNerush**

### CODE REVIEW

File: modules/ai/assets/js/gutenberg/index.js [hunk 1]
```diff
@@ -93,6 +84,21 @@ import { EditTextWithAi } from './edit-text-with-ai';
...
+						addGenerateExcerptWithAI();
+					}
+				}
+			}
+		} );
+
+		observer.observe( document.body, { childList: true, subtree: true } );
+		window.addEventListener( 'beforeunload', () => {
+			observer.disconnect();
+		} );
```
Comment (netanelavr) [inline modules/ai/assets/js/gutenberg/index.js:101]: can you please add a short explanation of why this is needed in the new version?

**Review by netanelavr — DISMISSED** [review]

File: modules/ai/assets/js/gutenberg/index.js (hunk 1 above)
Comment (DennisNerush) [reply modules/ai/assets/js/gutenberg/index.js:101]: The observer.observe(document.body, { childList: true, subtree: true }); line sets up a MutationObserver to monitor the entire document for changes in the DOM, specifically to detect when a popup (e.g., the excerpt editor) is added. This allows us to trigger necessary actions whenever the popup appears.

The window.addEventListener('beforeunload', () => { observer.disconnect(); }); line ensures that the observer is properly disconnected when the user navigates away from the current page. This prevents potential memory leaks and keeps our application efficient.

File: modules/ai/assets/js/gutenberg/index.js [hunk 2]
```diff
@@ -93,6 +84,21 @@ import { EditTextWithAi } from './edit-text-with-ai';
 			addAiIndicator( 'featured-image', addGenerateFeaturedImageWithAI );
 			addAiIndicatorToTextBlock( [ 'paragraph', 'heading' ] );
 		} );
+
+		const observer = new MutationObserver( ( mutationsList ) => {
+			for ( const mutation of mutationsList ) {
+				if ( 'childList' === mutation.type ) {
+					if ( document.querySelector( '.editor-post-excerpt' ) ) { // Replace with the actual class or ID of the popup
```
Comment (matipojo) [inline modules/ai/assets/js/gutenberg/index.js:91]:
```suggestion
					if ( document.querySelector( '.editor-post-excerpt' ) ) {
```

**Review by matipojo — DISMISSED** [review]

**Review by netanelavr — APPROVED** [review]

File: modules/ai/assets/js/gutenberg/index.js [hunk 3]
```diff
@@ -93,6 +84,21 @@ import { EditTextWithAi } from './edit-text-with-ai';
...
+			for ( const mutation of mutationsList ) {
+				if ( 'childList' === mutation.type ) {
+					if ( document.querySelector( '.editor-post-excerpt' ) ) {
+						addGenerateExcerptWithAI();
+					}
+				}
+			}
+		} );
+
+		observer.observe( document.body, { childList: true, subtree: true } );
```
Comment (matipojo) [inline modules/ai/assets/js/gutenberg/index.js:98]: Maybe observe only the sidebar instead of all changes in the body

Comment (DennisNerush) [reply modules/ai/assets/js/gutenberg/index.js:98]: it's being added to the body - not to the sidebar

**Review by matipojo — APPROVED** [review]

**Pull request merged**